### PR TITLE
Add Slots and Volunteer Basic Model

### PR DESCRIPTION
**Pull request by paridhi1603**
Please review

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "python.pythonPath": "/usr/local/bin/python3",
+    "python.pythonPath": "/usr/local/opt/python/bin/python3.7",
```
Comment (askvortsov1) [inline .vscode/settings.json:2]: Please revert this change: workspace-specific vscode settings should only be used for stuff thats applicable to all dev environments. Tbh, this file should be git ignored, and editor config should be used instead for shared settings.

File: requirements.txt [hunk 2]
```diff
@@ -4,3 +4,5 @@ djangorestframework
 django-cors-headers
 markdown       # Markdown support for the browsable API.
 django-filter  # Filtering support
+django-multiselectfield
+django-rest-knox
```
Comment (askvortsov1) [inline requirements.txt:8]: Why knox as opposed to JWT? Not inherently opposed, just curious.

File: src/cocoa_packs/settings.py [hunk 3]
```diff
@@ -31,9 +31,13 @@
     'phone_field',
     'rest_framework',
     'corsheaders',
+    'multiselectfield',
+    'knox',
 
     # startapp installations
     'donors',
+    'volunteers',
+
```
Comment (askvortsov1) [inline src/cocoa_packs/settings.py:40]: Unnecessary blank line

File: src/cocoa_packs/urls.py [hunk 4]
```diff
@@ -10,5 +10,10 @@
 urlpatterns = [
     path('api-auth/', include('rest_framework.urls')),
     path('admin/', admin.site.urls),
-    path('api/', include(router.urls)) 
+    path('api/', include(router.urls)),
+  
 ]
+
+
```
Comment (askvortsov1) [inline src/cocoa_packs/urls.py:17]: Please revert these blank line changes

File: src/volunteers/models.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from phone_field import PhoneField
+from multiselectfield import MultiSelectField
+
+
+class Slot(models.Model):
+    date = models.DateField(blank=True, default=None, null=True)
```
Comment (askvortsov1) [inline src/volunteers/models.py:8]: What does the slot represent? Do all slots have constant duration? Perhaps it should be datetime end and datetime start?

Also, I believe datetime fields are a thing: can we use them instead? So we'd probably end up with stsrt_datetome and end_datetime then

File: src/volunteers/models.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    def __str__(self):
+        return f"For {self.date} and time {self.time}"
+
+
+class Volunteer(models.Model):
+    CLEARANCES = (('PA Criminal History Report (PATCH)', 'PA Criminal History Report (PATCH)'),
+                  ('PA Child Abuse History Clearance (ChildLine)',
+                   'PA Child Abuse History Clearance (ChildLine)'),
+                  ('FBI Criminal Background Check', 'FBI Criminal Background Check'))
+    name = models.CharField(max_length=200)
```
Comment (askvortsov1) [inline src/volunteers/models.py:20]: Name, email, phone, etc are things not exclusively related to volunteers. Instead, this should act as a volunteer profile, which is linked via foreign key to a user. (in a one to one relationship!)

Name is already present on the user model, phone/email should either be added via a custom user model, or as a one to one Contact Profile linked via foreign key to the user.

File: src/volunteers/models.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    email = models.EmailField(max_length=200, blank=True, null=True)
+    phone = PhoneField(blank=True, null=True)
+    time_in = models.TimeField(blank=True, default=None, null=True)
+    time_out = models.TimeField(blank=True, default=None, null=True)
+    #slots= models.ForeignKey(Slot, on_delete=models.CASCADE, related_name= "slot_time")
```
Comment (askvortsov1) [inline src/volunteers/models.py:25]: This should probably be a many2many?

File: src/volunteers/models.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+class Volunteer(models.Model):
+    CLEARANCES = (('PA Criminal History Report (PATCH)', 'PA Criminal History Report (PATCH)'),
+                  ('PA Child Abuse History Clearance (ChildLine)',
+                   'PA Child Abuse History Clearance (ChildLine)'),
+                  ('FBI Criminal Background Check', 'FBI Criminal Background Check'))
+    name = models.CharField(max_length=200)
+    email = models.EmailField(max_length=200, blank=True, null=True)
+    phone = PhoneField(blank=True, null=True)
+    time_in = models.TimeField(blank=True, default=None, null=True)
```
Comment (askvortsov1) [inline src/volunteers/models.py:23]: Do volunteers always gave a constant schedule, each and every day? If not, these should be probably split out into a Assignment model (many2many)

File: src/volunteers/serializers.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
+from rest_framework import serializers
+from .models import Slot, Volunteer
+
+
+class SlotSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Slot
+        fields = '__all__'
```
Comment (askvortsov1) [inline src/volunteers/serializers.py:8]: Fields all is convenient, but bad practice and leads to data leaks. Although tedious, it's better to explicitly list out the attributes we're sending.

File: src/volunteers/models.py (hunk 8 above)
Comment (paridhi1603) [reply src/volunteers/models.py:23]: This is something we need to discuss with Peggy because according to Sign Up Genius, there is a schedule and people just sign up.

File: src/volunteers/models.py (hunk 7 above)
Comment (paridhi1603) [reply src/volunteers/models.py:25]: Alright! I'll look into that.

File: src/volunteers/models.py (hunk 5 above)
Comment (paridhi1603) [reply src/volunteers/models.py:8]: I am actually not very sure of the Slot class. I will work on it once we talk to Peggy.

File: requirements.txt (hunk 2 above)
Comment (paridhi1603) [reply requirements.txt:8]: I read online about using knox as a sign up and register feature. So was trying that out and currently working on it. if we can use JWT and it's better I'll probably do that

File: .vscode/settings.json (hunk 1 above)
Comment (paridhi1603) [reply .vscode/settings.json:2]: Alright!

File: requirements.txt (hunk 2 above)
Comment (askvortsov1) [reply requirements.txt:8]: Honestly knox is probably fine, I was just curious. Although if React has existing libraries for any of these authentication schemes that's definitely a plus (and probably the main heuristic for which to choose tbh)

File: src/volunteers/models.py (hunk 7 above)
Comment (askvortsov1) [reply src/volunteers/models.py:25]: Although re: https://github.com/Nittany-Data-Labs/Cocoa-Packs/pull/3#discussion_r449815290, it might make sense to confirm with Peggy first, yeah. As software engineers, our job is not only to convert the existing system into code, but also to make structural improvements to the system with possible. If we can improve something / do it better for our clients, why not do so?

File: src/volunteers/models.py (hunk 8 above)
Comment (askvortsov1) [reply src/volunteers/models.py:23]: Yeah in this case, we're probably going to want to move the stored-signing-up logic out (perhaps people are signing up for time slots?)

File: src/cocoa_packs/settings.py (hunk 3 above)
Comment (ashshekhar) [reply src/cocoa_packs/settings.py:40]: @paridhi1603 make sure to use Prettier before pushing your code. It'll take care of all this stuff.

File: src/cocoa_packs/urls.py (hunk 4 above)
Comment (ashshekhar) [reply src/cocoa_packs/urls.py:17]: @paridhi1603 make sure to use Prettier before pushing your code. It'll take care of all this stuff.

File: requirements.txt (hunk 2 above)
Comment (ashshekhar) [reply requirements.txt]: We wiill be using allauth
